lib/githubactions/token_source_test.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 Gravitational, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package githubactions
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestIDTokenSource_GetIDToken(t *testing.T) {
+	// TODO: Make this table driven & add more cases
+	requestToken := "foo-bar-biz"
+	idToken := "iam.a.jwt"
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// mimics the internal API accessible by Github Actions
+		require.Equal(t, http.MethodGet, r.Method)
```

Comment:
Requests of the server will run on a background goroutine, right? If so, it's not valid to call `require.Equal` here.

Reply:
38a335804